distribution/pom.xml

```diff
@@ -485,6 +485,36 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <!-- this profile adds/overrides few features of the 'apache-release'
+        profile in the parent pom. -->
+      <id>apache-release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.spotify</groupId>
+            <artifactId>dockerfile-maven-plugin</artifactId>
+            <version>1.4.3</version>
+            <executions>
+              <execution>
+                <id>docker-image</id>
+                <goals>
```

Comment:
Will it push without phase being specified? Example: `<phase>package</phase>`

Reply:
We can get rid of 'push' if you'd like. 'build' is good enough for most purposes. I'm not sure about 'package'.

Reply:
Here is how we can explicitly build, tag and push (any one action at at time):

Inside the distribution directory, executing:
`mvn -Papache-release dockerfile:build`
`mvn -Papache-release dockerfile:tag`
`mvn -Papache-release dockerfile:push`

Build and Tag happens on the user's system. Push pushes the image to the container registry on Docker Hub: https://hub.docker.com/r/drill/apache-drill-centos/tags/

On the root directory, one can simply run the below command to build and tag the container. Can be optionally pushed to Docker Hub separately. 
 `mvn clean install -Papache-release -DskipTests`

Reply:
Sounds like our Maven build is getting complex enough that it could benefit from a readme as well. (I've accumulated notes for how to generate code, how to rebuild protobufs, how to run single tests, etc. This information should be in a shared form...)

Reply:
Yeah, I agree - but there is no clean way to do it. We could add some notes in the dev docs later perhaps? Or do you have a different suggestion?